helix-term/src/commands.rs

```diff
@@ -3815,6 +3820,35 @@ pub mod insert {
         doc.apply(&transaction, view.id);
     }
 
+    pub fn insert_char_interactive(cx: &mut Context) {
+        let count = cx.count();
+
+        // need to wait for next key
+        cx.on_next_key(move |cx, event| {
+            match event {
+                KeyEvent {
+                    code: KeyCode::Char(ch),
+                    ..
+                } => {
+                    for _ in 0..count {
+                        insert::insert_char(cx, ch)
+                    }
+                }
+                KeyEvent {
+                    code: KeyCode::Enter,
+                    ..
+                } => {
+                    assert!(count == 1, "inserting multiple newlines not yet supported");
+                    insert_newline(cx)
+                }
+                KeyEvent {
+                    code: KeyCode::Tab, ..
+                } => insert_tab_impl(cx, count),
```

Comment:
These two branches can be cleaned up a bit if you `use crate::key;` in this block: `key!(Enter)` and `key!(Tab)`

Reply:
done